tools/proto2jsonschema/postgen.js

```diff
@@ -120,17 +126,55 @@ function getId(filepath) {
 }
 
 /**
- * Gets the CloudEvent#type represented in the JSON schema.
+ * Gets the cloudevent package represented in the JSON schema.
  * @param {string} filepath The input file path
  * @example filepath: /Documents/github/googleapis/google-cloudevents/jsonschema/google/events/cloud/audit/v1/LogEntryData.json
+ * @returns {string} The CloudEvent package for this file represents.
  * @example out: google.events.cloud.audit.v1
- * @returns {string} The CloudEvent ID for this file represents.
  */
-function getCloudEventType(filepath) {
+function getCloudEventPackage(filepath) {
   const removePrefix = filepath.split('jsonschema/')[1];
   const removeSuffix = removePrefix.substring(0, removePrefix.lastIndexOf("/"));
   return removeSuffix.replace(/\//g, '.');
 }
 
+/**
+ * Gets the CloudEvent properties from the corresponding `events.proto` file.
+ * @param {string} packageName The package name of the CloudEvent.
+ * @return {object} cloudevent The CloudEvent properties object.
+ * @return {string[]} cloudevent.types The CloudEvent type strings.
+ * @return {string} cloudevent.product The CloudEvent product.
+ */
+function getCloudEventProperties(packageName) {
+  const protoPath = packageName.split('.').join('/');
+  const eventPath = path.resolve(`${__dirname}/../../proto/${protoPath}/events.proto`);
+  const proto = protobufjs.loadSync(eventPath);
+  const protoAsJSON = proto.toJSON();
+
+  // Flatten the protobuf representation and look directly for these keys:
+  const CLOUD_EVENT_TYPE = '(google.events.cloud_event_type)';
+  const CLOUD_EVENT_PRODUCT = '(google.events.cloud_event_product)';
+  const flattenedProtoValueMap = flatten(protoAsJSON);
+  const flattenedProtoValueMapEntries = Object.entries(flattenedProtoValueMap);
+  
+  // Go through all the keys, add the type or product if found.
+  const types = [];
+  let product = '';
+  for (const [k, v] of flattenedProtoValueMapEntries) {
+    if (k.endsWith(CLOUD_EVENT_TYPE)) {
+      types.push(v);
+    }
+    if (k.endsWith(CLOUD_EVENT_PRODUCT)) {
+      product = v;
+    }
+  }
+
+  // Return types and product.
+  return {
+    types,
```

Comment:
Is "types" a clear enough name here? I'm just imagining someone loading the schema file without much context, and expecting "types" to mean data types rather than CloudEvent types. If we were to say "cloudEventTypes" it would potentially be clearer - but clunkier, certainly.

I'm on the fence (and this definitely doesn't block approval) - just something to think about before anything starts depending on the name, making it harder to change.

Reply:
I think I'll change to `cloudeventTypes` in the JSON schema to make it more clear.